src/plugins/vis_type_table/public/table_vis_fn.ts

```diff
@@ -69,18 +47,18 @@ export const createTableVisFn = (): ExpressionFunctionDefinition<
   },
   fn(input, args) {
     const visConfig = args.visConfig && JSON.parse(args.visConfig);
-    const convertedData = tableVisResponseHandler(input, visConfig.dimensions);
+    const convertedData = tableVisResponseHandler(input, visConfig);
```

Comment:
I'm curious - why did this change?

Reply:
I separate TableVisConfig from TableVisParams and  define TableVisConfig as 
```
export interface TableVisConfig extends TableVisParams {
  title: string;
  metrics: SchemaConfig[];
  buckets: SchemaConfig[];
  splitRow?: SchemaConfig[];
  splitColumn?: SchemaConfig[];
}
```
I think it is easier than using nested params.

Reply:
This is probably fine for now, but in general we'll want to standardize these configs. Agreed that the way `dimensions` are handled in general could use some streamlining. So happy to fix that as part of #2840 .